### ISSUE TITLE
TypeScript support

### ISSUE THREAD

**Issue by sam-goodwin** [smartsheet/smartsheet-javascript-sdk#34]
Is there a plan to add TypeScript to this project? 

Raw JS libraries are much harder to use because of the lack of types.

**Comment by andreasmarkussen** [comment]
Adding JSDocs could also help, and this is an easier way to get the Types you need.

**Comment by joekaiser** [comment]
I agree with @andreasmarkussen. JSDocs would be a quick path forward on this.

**Comment by christianblandford** [comment]
Would it be too crazy to create a [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) (@types) package for this instead of JSDoc? Happy to help get it started.

**Comment by christianblandford** [comment]
Working on creating types here and would welcome any help: https://github.com/christianblandford/smartsheet-javascript-sdk-types

I have all of the object models typed out, now need to do all of the methods/DTOs/etc...

**Comment by andreasmarkussen** [comment]
What is the difference? And can it be generated from JSDoc?

We use JSdoc at our project (not smartsheet) and this works just as fine. But perhaps this is inside the same repo

**Comment by spiceee** [comment]
Do you know if type declaration is in the roadmap at all? Smartsheet is pretty popular not to have TS support. 
Due to the lack of types, we gave up using the SDK for a project and are building a custom API wrapper.

**Comment by activescott** [comment]
There was a descent start on this done at smartsheet-platform/smartsheet-javascript-sdk#94 but it was never merged. The SDK hasn't moved much so it is probably still mostly valid. Happy to assist if someone wants a hand getting it updated to this repo.

**Comment by mattmeyerink** [comment]
We have added basic TypeScript support in the [latest release (v4.1.0)](https://github.com/smartsheet/smartsheet-javascript-sdk/releases/tag/v4.1.0) and will continue to move towards strict type support iteratively.
